File: hmkit-android/src/test/java/CryptoTest.java
// TODO: 24/07/2018 enable
/*
import com.highmobility.crypto.Crypto;
import com.highmobility.value.Bytes;
import com.highmobility.value.PrivateKey;
import com.highmobility.value.PublicKey;
import com.highmobility.value.Signature;

import org.junit.Test;

import static org.junit.Assert.assertTrue;

public class TestSignatures {
    // only work on linux
    @Test public void testSignatures() {
        PublicKey publicKey = new PublicKey
                ("***REMOVED***");
        PrivateKey privateKey = new PrivateKey
                ("***REMOVED***");

        assertTrue(true);

        Bytes data = new Bytes(new byte[]{0x02, 0x03});
        Signature sig = Crypto.sign(data, privateKey);

        assertTrue(Crypto.verify(data, sig, publicKey));

        publicKey = new PublicKey("***REMOVED***");
        assertTrue(Crypto.verify(data, sig, publicKey) == false);
    }
}
*/

File: hmkit-android/src/androidTest/java/com/highmobility/hmkit/CryptoTest.kt
/*
 * The MIT License
 *
 * Copyright (c) 2014- High-Mobility GmbH (https://high-mobility.com)
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in
 * all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 * THE SOFTWARE.
 */
package com.highmobility.hmkit;

import androidx.test.core.app.ApplicationProvider
import androidx.test.ext.junit.runners.AndroidJUnit4
import com.highmobility.crypto.value.PrivateKey
import com.highmobility.crypto.value.PublicKey
import com.highmobility.value.Bytes
import org.junit.Assert.assertTrue
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith

/**
 * Test crypto methods
 */
@RunWith(AndroidJUnit4::class)
class CryptoTest {

    private lateinit var kit:HMKit

    @Test fun testSignatures() {
        val crypto = kit.crypto
        var publicKey =
                PublicKey("A5A74048A85AC52A2E41DE5F9554C9CC36B6E3721EE8E8CE9169DC54192D17FD52C3BD1A4AE7F592756C083E17E54B7730965D99B238EB8D33B172DC35E32398")
        val privateKey =
                PrivateKey("1B8593D0478B9017C2427256AAEE25FF8A4E20EC6611AFE31D52B32CE0BECCA2")

        val data = Bytes("0203")
        val sig = crypto.sign(data, privateKey)

        assertTrue(crypto.verify(data, sig, publicKey))

        publicKey =
                PublicKey("A6A74048A85AC52A2E41DE5F9554C9CC36B6E3721EE8E8CE9169DC54192D17FD52C3BD1A4AE7F592756C083E17E54B7730965D99B238EB8D33B172DC35E32398")
        assertTrue(crypto.verify(data, sig, publicKey) == false)
    }

    @Before
    fun setup() {
        kit = HMKit.getInstance()
        kit.initialise(ApplicationProvider.getApplicationContext())
    }
}
